src/test/java/org/reaktivity/nukleus/maven/plugin/internal/generated/NestedFWTest.java

```diff
@@ -25,57 +26,67 @@
-    };
+        // Make sure the code is not secretly relying upon memory being initialized to 0
+        buffer.setMemory(0, buffer.capacity(), (byte) 0xab);
+    }
+
+    MutableDirectBuffer expected = new UnsafeBuffer(allocateDirect(100));
+    {
+        // Make sure the code is not secretly relying upon memory being initialized to 0
+        expected.setMemory(0, expected.capacity(), (byte) 0xab);
+    }
 
     @Rule
     public ExpectedException expectedException = ExpectedException.none();
 
+    @Test
+    public void shouldSetAllValues() throws Exception
+    {
+        int offset = 11;
+        int expectedLimit = setAllBufferValues(expected, offset);
+
+        NestedFW.Builder builder = setAllFieldsValues(nestedRW.wrap(buffer, offset, expectedLimit));
+
+        int limit = builder.build().limit();
+        assertEquals(expectedLimit, limit);
+        assertEquals(expected, buffer);
+
+        nestedRO.wrap(buffer,  offset,  limit);
+        assertAllValues(nestedRO);
+    }
+
     @Test
     public void shouldDefaultValues() throws Exception
     {
-        // Set an explicit value first in the same memory to make sure it really
-        // gets set to the default value next time round
-        int limit1 = nestedRW.wrap(buffer, 0, buffer.capacity())
-                .fixed4(40)
-                .flat(flat -> flat
-                    .fixed1(10)
-                    .fixed2(20)
-                    .string1("value1")
-                    .fixed3(30)
-                    .string2("value2")
-                 )
-                .fixed5(50)
-                .build()
-                .limit();
-        nestedRO.wrap(buffer,  0,  limit1);
-        assertEquals(40, nestedRO.fixed4());
-        assertEquals(20, nestedRO.flat().fixed2());
+        int offset = 11;
+        int expectedLimit = setAllRequiredBufferValues(expected, offset);
 
-        int limit2 = nestedRW.wrap(buffer, 0, 100)
-                .flat(flat -> flat
-                    .fixed1(10)
-                    .string1("value1")
-                    //.fixed3(30)
-                    .string2("value2")
-                )
-                .fixed5(50)
-                .build()
-                .limit();
-        nestedRO.wrap(buffer,  0,  limit2);
-        assertEquals(444, nestedRO.fixed4());
-        assertEquals(222, nestedRO.flat().fixed2());
-        assertEquals(333, nestedRO.flat().fixed3());
-        assertEquals(limit1, limit2);
+        NestedFW.Builder builder = setRequiredFields(nestedRW.wrap(buffer, offset, expectedLimit), INDEX_FIXED5);
+        int limit = builder.build().limit();
+
+        assertEquals(expectedLimit, limit);
+        assertEquals(expected, buffer);
+
+        nestedRO.wrap(buffer,  offset,  limit);
+        assertAllDefaultValues(nestedRO);
+    }
+
+    @Test(expected = IllegalArgumentException.class)
+    public void shouldFailToSetFixed4WithLowerValue()
+    {
+        nestedRW.wrap(buffer, 10, 10)
+                .fixed4(-1);
```

Comment:
Change shouldFailToBuildWhenFixed5IsNotSet to avoid "knowing about" flat idl:
```
        expectedException.expect(IllegalStateException.class);
        expectedException.expectMessage("fixed5");
        nestedRW.wrap(buffer, 0, 100)
            .flat(flat -> FlatFWTest.setRequiredFields(flat, Integer.MAX_VALUE)
            )
            .build();
```